### PR TITLE
Update the config and use flight_WEB_SUITE_domain

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
This PR does two things:

1. Updates to the latest version of `flight_configuration` and its usage, and
2. Fallback on `flight_WEB_SUITE_domain` to set the cookie domain.

### CODE REVIEW

File: api/etc/file-manager-api.yaml [hunk 1]
```diff
@@ -70,12 +70,17 @@
 #       contain a trailing newline.
 #
 # For security reasons, the command is not executed via a subshell. This means
-# all other string interplation/ variables have been disabled. Each argument is
+# all other string interpolation/ variables have been disabled. Each argument is
 # interpreted as a literal string and should not be quoted.
 #
-# Relative paths are expanded from the install directory.
+#>>path<<
```
Comment (benarmston) [inline api/etc/file-manager-api.yaml:76]: I'm not keen on this mechanism. See my comments on the omnibus builder PR for more details.

File: api/lib/flight.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      @root ||= if env.production? && ENV["flight_ROOT"].present?
+        File.expand_path(ENV["flight_ROOT"])
+      else
+        File.expand_path('..', __dir__)
+      end
+    end
+
+    def env
+      @env ||= ActiveSupport::StringInquirer.new(
+        ENV['RACK_ENV'].presence || "production"
```
Comment (benarmston) [inline api/lib/flight.rb:48]: Same as comment on `flight-login-api`.

File: api/lib/flight_file_manager/configuration.rb [hunk 3]
```diff
@@ -84,6 +85,7 @@ class ConfigError < StandardError; end
       {
         name: 'cloudcmd_cookie_domain',
         env_var: true,
+        default: ->() { RC['flight_WEB_SUITE_domain'] }
```
Comment (benarmston) [inline api/lib/flight_file_manager/configuration.rb:88]: Again, this can be `default: RC['flight_WEB_SUITE_domain']`.

**Review by benarmston — CHANGES_REQUESTED** [review]

File: api/etc/file-manager-api.yaml (hunk 1 above)
Comment (WilliamMcCumstie) [reply api/etc/file-manager-api.yaml]: 3dbd297f19e8a0e972c86f24b3a3cfd1d43d1c1a

File: api/config/boot.rb [hunk 4]
```diff
@@ -36,23 +35,36 @@
...
 
+require 'flight'
+
+Bundler.require(:default, :development) if Flight.env.development?
+
 require 'flight_file_manager'
 
 # Ensures the config has been loaded
-FlightFileManager.config.auth_decoder
+Flight.config.auth_decoder
```
[the remark this answers is not in the document]
Comment (WilliamMcCumstie) [reply api/config/boot.rb]: 3e038e3825f9de6194974fe118e1667823bade69

File: api/lib/flight.rb (hunk 2 above)
Comment (WilliamMcCumstie) [reply api/lib/flight.rb]: 971bc708514be831c599c57b4a5f4ab8e4fbc010

File: api/lib/flight_file_manager/configuration.rb (hunk 3 above)
Comment (WilliamMcCumstie) [reply api/lib/flight_file_manager/configuration.rb]: 0f9b9d430f768fbef5880db88c22e49511ff7be7

**Comment by benarmston** [comment]
This LGTM.

**Comment by benarmston** [comment]
Rebased on latest `master`.  LGTM.

**Pull request merged**